src/NSwag.MSBuild.CodeGeneration/NSwag.MSBuild.CodeGeneration.targets

```diff
@@ -0,0 +1,94 @@
+﻿<?xml version="1.0" encoding="utf-8" standalone="no"?>
+<Project>
+  <PropertyGroup>
+    <_NSwagcommand>$(NSwagExe)</_NSwagcommand>
+    <_NSwagcommand Condition="'$(MSBuildRuntimeType)' == 'Core'">$(NSwagExe_Core20)</_NSwagcommand>
+  </PropertyGroup>
+
+  <ItemDefinitionGroup>
+    <CurrentServiceProjectReference>
+      <GenerateNSwagDocumentOptions />
+      <Configuration />
+      <Platform />
+    </CurrentServiceProjectReference>
+    <CurrentServiceFileReference>
+      <GenerateNSwagCSharpOptions />
+      <GenerateNSwagTypeScriptOptions />
+    </CurrentServiceFileReference>
+  </ItemDefinitionGroup>
+
+  <!-- ServiceProjectReference support -->
+
+  <Target Name="GenerateNSwagDocument">
+    <ItemGroup>
+      <!-- @(CurrentServiceProjectReference) item group will never contain more than one item. -->
+      <CurrentServiceProjectReference Update="@(CurrentServiceProjectReference)">
+        <Command>$(NSwagExe)</Command>
```

Comment:
Shouldnt this also have a condition? Or is it overwritten by the next command?

Reply:
Later settings of the `%(Command)` metadata override earlier ones. This is just he default.